proposals/awslogs.md

```diff
@@ -0,0 +1,195 @@
+2017-09-28 22:32:12 [Wed Sep 27 22:32:12.300422 2017] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.10 (Debian) PHP/5.6.31 configured -- resuming normal operations
+2017-09-28 22:32:12 [Wed Sep 27 22:32:12.300456 2017] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'
+```
+
+### Implementation
+
+- The logs implementation will not include any pagination- the command will return all logs corresponding to the specified search. We expect most users will be piping the output of the command to save it to a file, so this should not be a problem.
+- If the user has not specified a log stream prefix in their task definition, then the command will fail and print an error message. Because without the log stream prefix set, we have no way of getting the logs for an individual task.
+- For performance reasons, the command will only pull from *a single log group*. If the customer has not configured all of their container definitions to use the same log group, then the command will fail with an error and tell the customer they must re-run the command with the `--container-name` argument. This way, only 1 log group needs to be queried.
+
+Work Flow:
+1.	User gives Task ID
+2.	Call Describe Tasks to get the TaskDef ARN (Skip this step if user provides Task Def)
+3.	Call Describe Task Definition to get the Container Definitions.
+4.	From Container Definitions, get the log configuration.
+5.	Create a list of log streams that correspond to the correct task for the log group.
+6.	Call FilterLogEvents for each log group to get the log events.
+7.	Print log events.
+
+
+## Compose Logs (Phase 2)
+Phase 2 will be implemented in the future when we have time, it is lower priority than Phase 1, and thus Phase 2 may not be implemented for some time. We welcome the contributions of any customer who wishes to help start implementation of Phase 2 sooner.
+
+### Configure Logs
+- Log configuration using the docker-compose file is already supported
+- Problem: Customer not required specify log stream prefix, however, we basically need log stream prefix to be specified because of how the ECS Agent sets the log stream name. If prefix is specified then it adds the container name and task ID to the log stream name (so we can use it to get the logs for each task). However, if a prefix is not specified, then the log stream name will for all intents and purposes be a random useless string (its an ID picked by the docker daemon on the instance, which from our point of view is meaningless).
+    - The log stream will be named like this (by the ECS Agent): `prefix-name/container-name/ecs-task-id`
+
+*Solution:* Existing ability to configure logs remains undisturbed, but add additional flag `--create-logs` that creates the necessary log group(s) in CloudWatch.
+- The log configuration from the docker compose file will be read
+- If user has not specified a log stream prefix, warn them that we are auto-setting it to a default value in their task definition.
```

Comment:
Why warn at all?  I think setting it is a reasonable behavior and we can just do that silently.